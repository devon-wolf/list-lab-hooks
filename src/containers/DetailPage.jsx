import React, { useState, useEffect } from 'react';
import CharacterDetail from '../components/CharacterDetail';
import { fetchTrekById } from '../services/starTrekFetch';

const DetailPage = ({ match }) => {
	const [loading, setLoading] = useState(false);
	const [character, setCharacter] = useState({});

	useEffect(() => {
		fetchTrekById(match.params.id)
			.then(result => {
				setCharacter(result);
				setLoading(false);
			})
			.catch(() => setLoading(false));
	}, []);

	return (
	<div>
		<h1>Detail page</h1>
		{loading
			? <p>Loading...</p>
			: <CharacterDetail 
				name={character.name}
				affiliation={character.affiliation}
				image={character.imageUrl}
				/>
		}
	</div>
)};

export default DetailPage;
