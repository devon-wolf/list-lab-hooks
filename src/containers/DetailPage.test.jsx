import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import DetailPage from './DetailPage';

describe('Detail page', () => {
  it('renders detail page', async () => {
    
	render(<DetailPage match={{ params: { id: '7' } }}/>);
	
	screen.getByText('Detail page');
	screen.getByText('Loading...');

	return waitFor(() => {
		const character = screen.getByLabelText('character details');
		expect(character).not.toBeEmptyDOMElement();
		expect(character).toMatchSnapshot();	
	});
		
	});
});
