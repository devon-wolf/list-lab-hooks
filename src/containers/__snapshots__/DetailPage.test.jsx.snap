// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail page renders detail page 1`] = `
<div
  aria-label="character details"
>
  <p>
    Mr. Spock
    : 
    Starfleet Command
  </p>
  <img
    alt="Mr. Spock"
    src="https://www.startrek.com/sites/default/files/styles/content_full/public/images/2019-07/8b10a9280bd46b8874af9b5cadec91d5.jpg?itok=3V9YqePf"
  />
</div>
`;
